kodi_addon_checker/report.py

```diff
@@ -0,0 +1,80 @@
+import inspect
+from abc import ABC
+
+PROBLEM = "ERROR"
+WARNING = "WARN"
+INFORMATION = "INFO"
+
+KODI_REPORTERS = {}
+
+
+class Report(object):
+    def __init__(self, artifact_name):
+        self.artifact_name = artifact_name
+        self.problem = 0
+        self.warning = 0
+        self.information = 0
+        self.reports = []
+
+    def log(self, report):
+        self.reports.append(report)
+        self.problem += report.problem
+        self.warning += report.warning
+        self.information += report.information
+
+    def __iter__(self):
+        return iter(self.reports)
+
+
+class Record(Report):
+    def __init__(self, artifact_name, message, start_line=-1, end_line=-1, start_char_position=-1,
```

Comment:
I'm not convinced by the naming of `artifact_name` here. Would be clearer as log level or similar.

Reply:
Fixed by renaming to `log_level`